src/functions/function-command/invoke-function.ts

```diff
@@ -0,0 +1,457 @@
+            {
+              value: parameters.invokeNamespace,
+              id: invokeFunctionID.invoke_namespace,
+              message: 'Provide namespace',
+            },
+            items,
+          );
+        }
+        if (parameters.invokePath !== undefined && !Number.isNaN(parameters.invokePath)) {
+          selectLocationValidation(
+            {
+              value: parameters.invokePath,
+              id: invokeFunctionID.invoke_path,
+              message: 'Provide path.',
+            },
+            items,
+          );
+        }
+        if (
+          parameters.invokeContextType !== undefined &&
+          !Number.isNaN(parameters.invokeContextType) &&
+          !parameters.invokeContextType?.trim()
+        ) {
+          inputFieldValidation(
+            {
+              value: parameters.invokeContextType,
+              id: invokeFunctionID.invoke_context_type,
+              message: 'Provide context type',
+            },
+            items,
+          );
+        }
+        if (parameters.invokeSource !== undefined && !Number.isNaN(parameters.invokeSource) && !parameters.invokeSource?.trim()) {
+          inputFieldValidation(
+            {
+              value: parameters.invokeSource,
+              id: invokeFunctionID.invoke_source,
+              message: 'Provide source',
+            },
+            items,
+          );
+        }
+        if (parameters.invokeType !== undefined && !Number.isNaN(parameters.invokeType) && !parameters.invokeType?.trim()) {
+          inputFieldValidation(
+            {
+              value: parameters.invokeType,
+              id: invokeFunctionID.invoke_type,
+              message: 'Provide type',
+            },
+            items,
+          );
+        }
+        if (
+          parameters.invokeDataText !== undefined &&
+          !Number.isNaN(parameters.invokeDataText) &&
+          !parameters.invokeDataText?.trim() &&
+          parameters.invokeDataMode === 'Text'
+        ) {
+          inputFieldValidation(
+            {
+              value: parameters.invokeDataText,
+              id: invokeFunctionID.invoke_data_text,
+              message: 'Provide data',
+            },
+            items,
+          );
+        }
+        if (
+          parameters.invokeDataFile !== undefined &&
+          !Number.isNaN(parameters.invokeDataFile) &&
+          !parameters.invokeDataFile?.trim() &&
+          parameters.invokeDataMode === 'File'
+        ) {
+          inputFieldValidation(
+            {
+              value: parameters.invokeDataFile,
+              id: invokeFunctionID.invoke_data_file,
+              message: 'Select file',
+            },
+            items,
+          );
+        }
+        return { items };
+      },
+    },
+  ],
+  buttons: [
+    {
+      id: BUTTONS.FINISH,
+      label: 'Invoke',
+    },
+  ],
+  workflowManager: {
+    canFinish(wizard: WebviewWizard, data: ParametersType): boolean {
```

Comment:
I'm really puzzled by seeing this. By reading the name of the method i guess that this is called to enable the `invoke` button by checking that all mandatory values have been filled by the user. The return is a boolean.
But then you try to get values from a map `invokeItemMap` which is not pre-filled anywhere (i only see a new Map<> but it doesn't get filled before executing the first time) and then you call the `invokeFunction` method which seems to create a new wizard and open it. Is the wizard recreated any time someone clicks something? How come is it so complicated? What's happening here? :laughing:

Reply:
so invoke UI get recreates when the user clicks on `local` or `remote` option and when the user changes the Mode `Text` or `File`.

Reply:
This recall me when i worked with VB.net, the UI was made in the background. BTW the workflowManager doesn't offer any other method to change the values you need? It still sounds weird to do it in `canFinish`

Reply:
I don't think that we have many options another option is I can move this code in validation part.

Reply:
Ok then let's keep the code like it is now. In validation it's the same for me. Resolving this conversation.